scheme/common/cca/realm/realm_extractor.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2024 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package cca
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/veraison/corim/comid"
+	"github.com/veraison/services/handler"
+)
+
+type RealmExtractor struct {
+	Scheme string
+}
+
+func (o RealmExtractor) RefValExtractor(rv comid.ReferenceValue) ([]*handler.Endorsement, error) {
+	var classAttrs RealmClassAttributes
+	var instAttrs RealmInstanceAttributes
+
+	if err := classAttrs.FromEnvironment(rv.Environment); err != nil {
+		return nil, fmt.Errorf("could not extract Realm class attributes: %w", err)
+	}
+
+	if err := instAttrs.FromEnvironment(rv.Environment); err != nil {
+		return nil, fmt.Errorf("could not extract Realm instance attributes: %w", err)
+	}
+
+	// Measurements are encoded in a measurement-map of a CoMID
+	// reference-triple-record.  Since a measurement-map can encode one or more
+	// measurements, a single reference-triple-record can carry as many
+	// measurements as needed. Hence for a Realm Instance, all the measurements
+	// which comprise of both the "RIM" & "REM" measurements are carried in an
```

Comment:
It's difficult to grasp the point of this comment. Please condense it to convey the essential information needed to understand the code.